com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -852,10 +899,8 @@ bool CanCopyDepth(ref CameraData cameraData)
             bool supportsDepthTarget = RenderingUtils.SupportsRenderTextureFormat(RenderTextureFormat.Depth);
             bool supportsDepthCopy = !msaaEnabledForCamera && (supportsDepthTarget || supportsTextureCopy);
 
-            // TODO:  We don't have support to highp Texture2DMS currently and this breaks depth precision.
-            // currently disabling it until shader changes kick in.
-            //bool msaaDepthResolve = msaaEnabledForCamera && SystemInfo.supportsMultisampledTextures != 0;
-            bool msaaDepthResolve = false;
+            bool msaaDepthResolve = msaaEnabledForCamera && SystemInfo.supportsMultisampledTextures != 0;
+
```

Comment:
this will now introduce an MSAA depth copy on mobile/TBDR platforms when MSAA > 1 (instead of the current non-AA depth prepass) which is quite a big hit performance wise. Looks like a regression

to reproduce the issue:
enable MSAA, tick "Depth Texture". The depth prepass should now be replaced by a MSAA'd copy

Reply:
I have added an additional condition that hasHiddenSurfaceRemovalOnGPU must be false however that will still let non HSR tiled GPU's slip through the cracks. This is another reason to expose hasTiledGPU to SystemInfo.
I think I'll just set this back to false until we can guarantee there's no perf regression for tiled hardware.

Reply:
Sounds good thanks!
In general I think it is good to enable the copy and remove depth prepass when MSAA > 1 on mobile platforms.

But since the change is not directly related to the depth priming, I agree it's safer to do it in a separate PR so all platforms can be tested for potential regressions. I am planning to look into that once some engine PRs (in progress) with new depth resolve API will land